openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -39,9 +39,16 @@ def process_request(self, request):
             preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
             lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
 
+            prefixes = [prefix.split("-")[0] for prefix in system_released_languages]
             # Setting the session language to the browser language, if it is supported.
             for browser_lang in lang_headers:
                 if browser_lang in system_released_languages:
                     if request.session.get(LANGUAGE_SESSION_KEY, None) is None:
                         request.session[LANGUAGE_SESSION_KEY] = unicode(browser_lang)
                     break
+                else:
+                    if browser_lang in prefixes:
+                        browser_lang = system_released_languages[prefixes.index(browser_lang)]
+                        if request.session.get(LANGUAGE_SESSION_KEY, None) is None:
+                            request.session[LANGUAGE_SESSION_KEY] = unicode(browser_lang)
+                        break
```

Comment:
Please "DRY" (Don't Repeat Yourself) the above code.  In particular, there is duplicated logic in L44-54.  For example, you can rewrite the code to:

```
for browser_lang in lang_headers:
    if browser_lang in system_released_languages:
        pass
    elif browser_lang in prefixes:
        browser_lang = system_released_languages[prefixes.index(browser_lang)]
    else:
        continue
    if request.session.get(LANGUAGE_SESSION_KEY, None) is None:
        request.session[LANGUAGE_SESSION_KEY] = unicode(browser_lang)
    break
```